base/pcre.jl

```diff
@@ -144,20 +150,25 @@ free_jit_stack(stack) =
 free_match_context(context) =
     ccall((:pcre2_match_context_free_8, PCRE_LIB), Cvoid, (Ptr{Cvoid},), context)
 
-function err_message(errno)
-    buffer = Vector{UInt8}(undef, 256)
-    ccall((:pcre2_get_error_message_8, PCRE_LIB), Cvoid,
-          (UInt32, Ptr{UInt8}, Csize_t), errno, buffer, sizeof(buffer))
-    GC.@preserve buffer unsafe_string(pointer(buffer))
+function err_message(errno::Integer)
+    buffer = Vector{UInt8}(undef, 1024)
+    ret = ccall((:pcre2_get_error_message_8, PCRE_LIB), Cint,
+                (Cint, Ptr{UInt8}, Csize_t), errno, buffer, length(buffer))
+    ret == ERROR_BADDATA && error("PCRE error: invalid errno ($errno)")
+    # TODO: seems like there should be a better way to get this string
```

Comment:
`String(buffer)` should work.

Reply:
Won't that include the NUL byte and following data?

Reply:
Oh, right, the vector would need to be resized first.

Reply:
I'm fine with this, I had a slight worry about the memory leaking though.